File_Adapter/AdapterActions/Push.cs

```diff
@@ -60,13 +57,9 @@ public override List<object> Push(IEnumerable<object> objects, string tag = "",
 
             CreateFileAndFolder(pushType);
 
-            if (objectsToPush.Count() != objects.Count())
-                Engine.Reflection.Compute.RecordWarning("The file adapter can currently only be used with BHoMObjects." + Environment.NewLine +
-                    "If you want to push non-BHoMobject, specify a push config with the option `WrapNonBHoMObject` set to true.");
-
-            bool success = this.FullCRUD(objectsToPush, pushType, tag, actionConfig);
+            bool success = this.FullCRUD(objects.OfType<IBHoMObject>(), pushType, tag, actionConfig);
 
-            return success ? objectsToPush.Cast<object>().ToList() : new List<IObject>().Cast<object>().ToList();
+            return success ? objects.ToList() : new List<IObject>().Cast<object>().ToList();
```

Comment:
not something changed here, but cant it just be:

```suggestion
            return success ? objects.ToList() : new List<object>();
```

not sure I get the casting and to list thing there? Might be leftover from previous iterations?